### PR TITLE
Ensure at least one peer is connected before starting cfheader sync.

### PR DESCRIPTION
**Pull request by roeierez**
This PR is a suggested way to reduce about 3 seconds (with default configuration) from neutrino startup time on some cases.
From within BlockManager the function cfHandler is called right on start regardless if there is at least one peer connected.
Since this function query the peers it causes the first query to fail with high probability and retry after a waiting period of QueryTimeout interval (3 seconds by default).
I have tested the startup on mobile devices and found that the first peer connection is usually established in less than 1 second, which creates a waste of 2-3 seconds.
My suggestion is to start the cfHandler after there is at least one peer, making sure the query has a target.

### CODE REVIEW

File: neutrino.go [hunk 1]
```diff
@@ -1115,6 +1117,11 @@ func (s *ChainService) handleAddPeerMsg(state *peerState, sp *ServerPeer) bool {
 		state.outboundPeers[sp.ID()] = sp
 	}
 
+	//close firstPeerConnect channel so blockManager will be notified
+	if s.firstPeerConnect != nil {
+		close(s.firstPeerConnect)
+		s.firstPeerConnect = nil
+	}
```
Comment (aakselrod) [inline neutrino.go]:
```suggestion
	}
	
```

**Comment by aakselrod** [comment]
Other than a small whitespace issue, this LGTM.

File: neutrino.go (hunk 1 above)
Comment (roeierez) [reply neutrino.go]: fixed

**Comment by roeierez** [comment]
squashed.

File: neutrino.go [hunk 2]
```diff
@@ -1115,6 +1117,12 @@ func (s *ChainService) handleAddPeerMsg(state *peerState, sp *ServerPeer) bool {
 		state.outboundPeers[sp.ID()] = sp
 	}
 
+	// Close firstPeerConnect channel so blockManager will be notified
```
Comment (halseth) [inline neutrino.go]: nit: period at end.

File: blockmanager.go [hunk 3]
```diff
@@ -259,7 +259,16 @@ func (b *blockManager) Start() {
 	log.Trace("Starting block manager")
 	b.wg.Add(2)
 	go b.blockHandler()
-	go b.cfHandler()
+	go func() {
+		defer b.wg.Done()
+
+		select {
+		case <-firstPeerConnect:
```
Comment (halseth) [inline blockmanager.go]: this should have a comment.

File: blockmanager.go [hunk 4]
```diff
@@ -411,7 +420,6 @@ func (b *blockManager) cfHandler() {
 	// done.
 	defer func() {
```
Comment (halseth) [inline blockmanager.go]: function closure not needed anymore

File: blockmanager.go [hunk 5]
```diff
@@ -259,7 +259,16 @@ func (b *blockManager) Start() {
 	log.Trace("Starting block manager")
 	b.wg.Add(2)
 	go b.blockHandler()
-	go b.cfHandler()
+	go func() {
+		defer b.wg.Done()
+
+		select {
+		case <-firstPeerConnect:
+			b.cfHandler()
```
Comment (halseth) [inline blockmanager.go]: I would also add debug logs `"Waiting for peer connection..."` and `"Peer connected, starting cfHandler..."`

Comment (roeierez) [reply blockmanager.go]: good suggestion, added.

File: blockmanager.go (hunk 3 above)
Comment (roeierez) [reply blockmanager.go]: added.

File: blockmanager.go (hunk 4 above)
Comment (roeierez) [reply blockmanager.go]: I removed the outdated comment.
I thought of leaving the rest for leaving the  logging behavior as it was. Do you suggest to remove that?

File: neutrino.go (hunk 2 above)
Comment (roeierez) [reply neutrino.go]: fixed

File: blockmanager.go [hunk 6]
```diff
@@ -259,7 +259,21 @@ func (b *blockManager) Start() {
 	log.Trace("Starting block manager")
 	b.wg.Add(2)
 	go b.blockHandler()
-	go b.cfHandler()
+	go func() {
+		defer b.wg.Done()
+
+		log.Debug("Waiting for peer connection...")
+
+		// Before starting the cfHandler we want to make sure we are connected
```
Comment (halseth) [inline blockmanager.go]: line looks long?

File: blockmanager.go (hunk 4 above)
Comment (halseth) [reply blockmanager.go]: Oh, is just meant that this cane be `defer log.Trace("Committed filter header handler done")` now :)

File: blockmanager.go (hunk 6 above)
Comment (roeierez) [reply blockmanager.go]: 79 characters, wrapping should start at 80 right?

File: blockmanager.go (hunk 4 above)
Comment (roeierez) [reply blockmanager.go]: sure thing! fixed.

File: blockmanager.go [hunk 7]
```diff
@@ -259,7 +259,21 @@ func (b *blockManager) Start() {
...
+		// with at least one peer.
+		select {
+		case <-firstPeerConnect:
+			log.Debug("Peer connected, starting cfHandler...")
```
Comment (halseth) [inline blockmanager.go]: nit: `log.Debug("Peer connected, starting cfHandler.")`

File: blockmanager.go [hunk 8]
```diff
@@ -259,7 +259,21 @@ func (b *blockManager) Start() {
...
+			b.cfHandler()
```
Comment (halseth) [inline blockmanager.go]: style nit: this select can be written

```golang
select {
	case <-firstPeerConnect:
	case <-b.quit:
		return
	}
}

log.Debug("Peer connected, starting cfHandler...")
b.cfHandler()
```

File: blockmanager.go (hunk 7 above)
Comment (roeierez) [reply blockmanager.go]: fixed

File: blockmanager.go (hunk 8 above)
Comment (roeierez) [reply blockmanager.go]: fixed

**Pull request merged**